src/openrct2/platform/Platform.Linux.cpp

```diff
@@ -271,6 +271,45 @@ namespace Platform
         }
         return MeasurementFormat::Metric;
     }
+
+    std::string GetSteamPath()
+    {
+        const char* steamRoot = getenv("STEAMROOT");
+        if (steamRoot != nullptr)
```

Comment:
You could simplify with

```c++
        if (auto steamRoot = getenv("STEAMROOT"))
```

Or if you want to be explicit:
```c++
        if (auto steamRoot = getenv("STEAMROOT"); steamRoot != nullptr)
```

That would also reduce the scope of the variable to only where it is needed. This would also apply to a lot more `if`s in this function.

Reply:
I don’t see that as a simplification. On the contrary, I think it makes it much more difficult to read.

Reply:
I also would prefer this, we don't need to extend the lifetime of variables any longer than needed.